src/error.rs

```diff
@@ -0,0 +1,53 @@
+use std::ffi::OsString;
+use std::path::PathBuf;
+use thiserror::Error;
+
+pub type Result<T> = std::result::Result<T, Error>;
+
+#[derive(Debug, Error)]
+pub enum Error {
```

Comment:
For now is good, but in the future we probably want to split certain error types at the mod level. Keeping a 'generic' one for all the crate should be good for now though.

Reply:
Agreed. I think most of the existing errors would fall under configuration, but the configuration is very free form and split up at the moment.